apps/binding.go

```diff
@@ -1,39 +1,128 @@
 package apps
 
+// Binding is the main way for an App to attach its functionality to the
+// Mattermost UI.
+//
+// An App returns the bindings in response to the "bindings" call, that it must
+// implement, and can customize in its Manifest. For each context in which it is
+// invoked, the bindings call returns a tree of app's bindings, organized by the
+// top-level location.
+//
+// Top level bindings need to define:
+//  location - the top-level location to bind to, e.g. "post_menu".
+//  bindings - an array of bindings
+//
+// /post_menu bindings need to define:
+//  location - Name of this location. The whole path of locations will be added in the context.
+//  icon - optional URL or path to the icon
+//  label - Text to show in the item
+//  call - Call to perform.
+//
+// /channel_header bindings need to define:
+//  location - Name of this location. The whole path of locations will be added in the context.
+//  icon - optional URL or path to the icon
+//  label - text to show in the item on mobile and webapp collapsed view.
+//  hint - text to show in the tooltip.
+//  call - Call to perform.
+//
+// /command bindings can define "inner" subcommands that are collections of more
```

Comment:
Commands just got a bit more complex. We have the "inner" commands, the "outer" commands, and the new "basecommand", which may provide description and hint (and in the future icon).

Reply:
fixed